tests/Core/Tokenizer/EnumCaseTest.php

```diff
@@ -0,0 +1,142 @@
+<?php
+/**
+ * Tests converting enum "case" to T_ENUM_CASE.
+ *
+ * @author    Jaroslav Hanslík <kukulich@kukulich.cz>
+ * @copyright 2021 Squiz Pty Ltd (ABN 77 084 670 600)
+ * @license   https://github.com/squizlabs/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ */
+
+namespace PHP_CodeSniffer\Tests\Core\Tokenizer;
+
+use PHP_CodeSniffer\Tests\Core\AbstractMethodUnitTest;
+
+class EnumCaseTest extends AbstractMethodUnitTest
+{
+
+
+    /**
+     * Test that the enum "case" is converted to T_ENUM_CASE.
+     *
+     * @param string $testMarker  The comment which prefaces the target token in the test file.
+     * @param string $testContent The token content to expect.
+     *
+     * @dataProvider dataEnumCases
+     * @covers       PHP_CodeSniffer\Tokenizers\PHP::processAdditional
+     *
+     * @return void
+     */
+    public function testEnumCases($testMarker, $testContent)
+    {
+        $tokens = self::$phpcsFile->getTokens();
+
+        $enum = $this->getTargetToken($testMarker, [T_ENUM_CASE, T_CASE], $testContent);
+
+        $this->assertSame(T_ENUM_CASE, $tokens[$enum]['code']);
+        $this->assertSame('T_ENUM_CASE', $tokens[$enum]['type']);
```

Comment:
I believe it would be good idea to ensure that the `scope_owner`, `scope_opener` and `scope_closer` keys are **_not_** set for the found token.
Again, this is similar to what is done in the `DefaultKeywordTest::testMatchDefault()` test method.

Might it be an idea to transform this complete test along the same lines as the `DefaultKeywordTest` ?
I.e. `CaseKeywordTest` and have detailed tests for both the `enum` `case` keyword (including making sure no scope is set) as well as the `switch` `case` keyword (including making sure scope **is** set)  and have a separate set of tests in the same file for the `case` keyword being used when not a keyword ?

Reply:
> I believe it would be good idea to ensure that the scope_owner, scope_opener and scope_closer keys are not set for the found token.

Added.